cmd/swarm/swarm-snapshot/create.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2018 The go-ethereum Authors
+// This file is part of go-ethereum.
+//
+// go-ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// go-ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/cmd/utils"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/p2p/simulations"
+	"github.com/ethereum/go-ethereum/p2p/simulations/adapters"
+	"github.com/ethereum/go-ethereum/swarm/network"
+	"github.com/ethereum/go-ethereum/swarm/network/simulation"
+	cli "gopkg.in/urfave/cli.v1"
+)
+
+var serviceFuncs = adapters.Services{
+	"discovery": newService,
+}
+
+const testMinProxBinSize = 2
+const NoConnectionTimeout = 1
+
+var discoveryEnabled = true
+
+func init() {
+	adapters.RegisterServices(serviceFuncs)
+}
+
+func create(ctx *cli.Context) error {
+	log.PrintOrigins(true)
+	log.Root().SetHandler(log.LvlFilterHandler(log.Lvl(verbosity), log.StreamHandler(os.Stdout, log.TerminalFormat(true))))
+	if len(ctx.Args()) < 1 {
+		return errors.New("argument should be the filename to verify or write-to")
+	}
+	filename = ctx.Args()[0]
+	err := ResolvePath()
+	if err != nil {
+		return err
+	}
+	err = discoverySnapshot(10, adapters.NewSimAdapter(serviceFuncs))
+	if err != nil {
+		utils.Fatalf("Simulation failed: %s", err)
+	}
+
+	return err
+}
+
+func discoverySnapshot(nodes int, adapter adapters.NodeAdapter) error {
+	// create network
+	net := simulations.NewNetwork(adapter, &simulations.NetworkConfig{
+		ID:             "0",
+		DefaultService: "discovery",
+	})
+	defer net.Shutdown()
+	ids, err := net.AddNodes(nodes)
```

Comment:
Currently this call means no alternative but fully random nodes. Perhaps, though, we will need a way to control certain constraints of addresses for simulations in the future.

Reply:
once @frncmx moves the `node.go` file we can propagate this fully

Reply:
> Perhaps, though, we will need a way to control certain constraints of addresses for simulations in the future. Should we suggest a proper interface for such parameters already?

I would say [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it): "Always implement things when you actually need them, never when you just foresee that you need them."